File: my-folder/0004-median-of-two-sorted-arrays/solution.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int k = nums1.length + nums2.length;
        int[] nums3 = new int[k];
        for (int i = 0; i < nums1.length; ++i)
        {
            nums3[i] = nums1[i];
        }
        for (int i = 0; i < nums2.length; ++i)
        {
            nums3[i + nums1.length] = nums2[i];
        }
        Arrays.sort(nums3);
        if (nums3.length % 2 != 0)
        {
            return nums3[k / 2];
        }
        else
        {
            return (nums3[k / 2] + nums3[(k / 2) - 1]) / 2.0;
        }
    }
}

File: my-folder/0004-median-of-two-sorted-arrays/solution.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> nums;
        int ptr1 = 0, ptr2 = 0;
        while ((ptr1 < nums1.size()) && (ptr2 < nums2.size())) {
            if (nums1[ptr1] < nums2[ptr2]) {
                nums.push_back(nums1[ptr1]);
                ++ptr1;
            }
            else {
                nums.push_back(nums2[ptr2]);
                ++ptr2;
            }
        }
        while (ptr1 < nums1.size()) {
            nums.push_back(nums1[ptr1]);
            ++ptr1;
        }
        while (ptr2 < nums2.size()) {
            nums.push_back(nums2[ptr2]);
            ++ptr2;
        }

        int size = nums.size();
        cout << size << endl;
        if (size % 2 == 0) {
            return (nums[size / 2] + nums[size / 2 - 1]) / 2.0;
        }
        else {
            return nums[size / 2];
        }
    }
};
